### PR TITLE
feat(rds): add support for update and backup properties to Cluster instances

### LINKED ISSUE #9926: [aws-rds] Supports defining variables for a CfnDBInstance in a DatabaseCluster
Add the following properties in DatabaseCluster.
```
autoMinorVersionUpgrade: boolean
allowMajorVersionUpgrade: boolean
deleteAutomatedBackups: boolean
enablePerformanceInsights: boolean
```
The most important one is `autoMinorVersionUpgrade`.

#### `autoMinorVersionUpgrade`
When the database instance is automatically upgrade by AWS, drift is detected between cdk(cloudformation) and the actual resource, and the upgrade cannot be performed with cdk.
To work around this and deploy the cloudformation, you need to remove the RDS from the management target and re-include it. cdk does not support this.
We should be able to manage the migration in our own time.

#### Others (`allowMajorVersionUpgrade`, `deleteAutomatedBackups`, `enablePerformanceInsights`)
I'd like to set these parameters as well.

### Proposed Solution

1. Define the properties below
https://github.com/aws/aws-cdk/blob/3c116f3b5854726bef610e425fc35f2d2b50a179/packages/%40aws-cdk/aws-rds/lib/props.ts#L10
2. Pass properties to CfnDBInstance
https://github.com/aws/aws-cdk/blob/3c116f3b5854726bef610e425fc35f2d2b50a179/packages/%40aws-cdk/aws-rds/lib/cluster.ts#L590

### Other

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by hixi-hyi**
fixes #9926

Added the following parameters to DatabaseCluster.
* AutoMinorVersionUpgrade
* AllowMajorVersionUpgrade
* DeleteAutomatedBackups

#10092 as a reference, only defined simple parameters.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by hixi-hyi** [comment]
If this policy is acceptable, I can also add these parameters to DatabaseInstance.

File: packages/@aws-cdk/aws-rds/lib/props.ts [hunk 1]
```diff
@@ -63,6 +63,25 @@ export interface InstanceProps {
    * @default - default master key
    */
   readonly performanceInsightEncryptionKey?: kms.IKey;
+
+  /**
+   * Whether to enable automatic upgrade of minor version for the DB instance.
+   *
+   * @default - true
+   */
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/props.ts:71]: Should have an empty line after this line.

File: packages/@aws-cdk/aws-rds/lib/props.ts [hunk 2]
```diff
@@ -63,6 +63,25 @@ export interface InstanceProps {
...
+  readonly autoMinorVersionUpgrade?: boolean;
+  /**
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/props.ts:73]: Should have an empty line before this line.

File: packages/@aws-cdk/aws-rds/lib/props.ts [hunk 3]
```diff
@@ -63,6 +63,25 @@ export interface InstanceProps {
...
+   * Whether to allow upgrade of major version for the DB instance.
+   *
+   * @default - true
+   */
+  readonly allowMajorVersionUpgrade?: boolean;
+  /**
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/props.ts:79]: Should have an empty line before this line.

File: packages/@aws-cdk/aws-rds/lib/props.ts [hunk 4]
```diff
@@ -63,6 +63,25 @@ export interface InstanceProps {
...
+   * Whether to enable automatic upgrade of minor version for the DB instance.
+   *
+   * @default - true
+   */
+  readonly autoMinorVersionUpgrade?: boolean;
+  /**
+   * Whether to allow upgrade of major version for the DB instance.
+   *
+   * @default - true
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/props.ts:76]: is this really `true` by default? The [documentation says](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-rds-database-instance.html#cfn-rds-dbinstance-autominorversionupgrade):

```
Constraints: Major version upgrades must be allowed when specifying a value for the EngineVersion parameter that is a different major version than the DB instance's current version.
```

I guess that suggests that the default is, in fact, `false`...?

**Review by skinny85 — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution @hixi-hyi ! Some small comments, and one question.

File: packages/@aws-cdk/aws-rds/lib/props.ts (hunk 4 above)
Comment (hixi-hyi) [reply packages/@aws-cdk/aws-rds/lib/props.ts:76]: Surely the default value for `AllowMajorVersionUpgrade` may be false.
Reading the documentation, this parameter may not be used routinely and is only needed when changing instance types, but I don't understand exactly what it is. (There was no value for `AllowMajorVersionUpgrade` in the results of `aws rds describe-instances`)

The `@default` value is currently written from the Cloudformation documentation.
However, I haven't programmatically set it, and its behavior is dependent on Cloudformation.

Should such a parameter be written as `@default None`? Or should I explicitly set true or false in the cdk?

I'd like to hear your opinion.

File: packages/@aws-cdk/aws-rds/lib/props.ts (hunk 1 above)
Comment (hixi-hyi) [reply packages/@aws-cdk/aws-rds/lib/props.ts:71]: Perhaps it's not necessary.

File: packages/@aws-cdk/aws-rds/lib/props.ts (hunk 4 above)
Comment (skinny85) [reply packages/@aws-cdk/aws-rds/lib/props.ts:77]: If not setting it and setting it to `false` provide the same result (as I believe it does in this case), it should be `@default false`.

File: packages/@aws-cdk/aws-rds/lib/props.ts (hunk 1 above)
Comment (skinny85) [reply packages/@aws-cdk/aws-rds/lib/props.ts:71]: Yep, I think this was referring to a different line in the original diff 🙂. Please disregard.

**Comment by skinny85** [comment]
Hey @hixi-hyi ,

looks like some trailing spaces snuck into the file:

```
@aws-cdk/aws-rds: /codebuild/output/src654474608/src/github.com/aws/aws-cdk/packages/@aws-cdk/aws-rds/lib/props.ts
@aws-cdk/aws-rds:   73:1  error  Trailing spaces not allowed  no-trailing-spaces
@aws-cdk/aws-rds:   80:1  error  Trailing spaces not allowed  no-trailing-spaces
```

Would you mind updating the PR (and also inclding the change discussed in https://github.com/aws/aws-cdk/pull/10324#discussion_r493672482 )? Thanks!

**Review by skinny85 — CHANGES_REQUESTED** [review]
Putting in "Request changes" awaiting @hixi-hyi 's new iteration to remove this from my ToDo list.

File: packages/@aws-cdk/aws-rds/lib/props.ts (hunk 4 above)
Comment (hixi-hyi) [reply packages/@aws-cdk/aws-rds/lib/props.ts]: I understood. Thank you for your reply.
In fact, `AllowMajorUpgrade` is default false, and it seems to be necessary to add this parameter when upgrading the major version.

**Comment by hixi-hyi** [comment]
@skinny85 
I forgot to run `yarn lint` on gitpod, which I did. 
I'm very sorry, but I'd like to know if there's anything else that needs to be addressed.

**Review by skinny85 — APPROVED** [review]
> @skinny85
> I forgot to run `yarn lint` on gitpod, which I did.
> I'm very sorry, but I'd like to know if there's anything else that needs to be addressed.

Everything is great @hixi-hyi , thanks so much for your contribution!

**Review by skinny85 — APPROVED** [review]

**Comment by hixi-hyi** [comment]
Thanks for the merge and the fix testcode :D

**Comment by skinny85** [comment]
> Thanks for the merge and the fix testcode :D

My pleasure, thanks for the contribution!

**Pull request merged**